src/modules/form/components/FormNavigationContainer.tsx

```diff
@@ -0,0 +1,90 @@
+import { Divider, Grid, Paper, Stack, Typography } from '@mui/material';
+import { ReactNode } from 'react';
+import SkipToContentButton from '@/components/elements/SkipToContentButton';
+import {
+  CONTEXT_HEADER_HEIGHT,
+  STICKY_BAR_HEIGHT,
+} from '@/components/layout/layoutConstants';
+import { useIsMobile } from '@/hooks/useIsMobile';
+import FormStepper from '@/modules/form/components/FormStepper';
+import { FormItem } from '@/types/gqlTypes';
+
+export interface FormNavigationProps {
+  navTitle?: string | null;
+  navItems?: FormItem[];
+  navScrollOffset?: number;
+  contentsBelowNavigation?: ReactNode;
+  useUrlHash?: boolean;
+  children: ReactNode;
+}
+const FormNavigationContainer = ({
+  navItems,
+  navTitle = 'Form Navigation',
+  navScrollOffset = STICKY_BAR_HEIGHT + CONTEXT_HEADER_HEIGHT,
+  contentsBelowNavigation,
+  useUrlHash = true,
+  children,
+}: FormNavigationProps) => {
+  const isTiny = useIsMobile('sm');
```

Comment:
I had some hopes of making this depend on `useDashboardContext`'s `desktopNavIsOpen` for improved mobile-friendliness, but I didn't get that working in the timebox I gave myself for this refactor. The form stepper is moderately more mobile-friendly now though.